File: cpp/problem009.cpp
/*
    Copyright (c) 2015 - buyuk.dev@gmail.com
    ----------------------------------------
    Compute Pythagorean triplet for which a+b+c == 1000
*/

#include <iostream>
#include <vector>
#include <string>
#include <list>
#include <map>
#include <algorithm>
#include <cmath>


#define MAX(a,b) (a<b?b:a)
#define MIN(a,b) (a<b?a:b)



using namespace std;



int main(int argc, char* argv[]) 
{
    ios_base::sync_with_stdio(0);

    for (int a=1; a<1000; ++a) {
        for (int b=a; b<1000; ++b) {
            int c = sqrt(a*a + b*b);
            if (a+b+c == 1000 && a*a + b*b == c*c) {
                cout << a << " " << b << " " << c << endl;
                break;
            }
        }
    }
    
    return 0;
}

File: python/problem009.py
#
# Copyright (c) 2016 - buyuk.dev@gmail.com
# ----------------------------------------
# Special Pythagorean Triplet
# 

    
def isPythagorean(a, b, c):

    return a*a + b*b == c*c
    

def tripletGenerator(n):

    for a in range(1, n // 2 + 1):
        b = a
        while b <= n - (a + b): 
            c = n - (a + b)
            yield (a, b, c)
            b += 1

            
if __name__ == "__main__":

    n = 1000
    result = 0
        
    for a,b,c in tripletGenerator(n):
    
        if isPythagorean(a, b, c):  
            print "Pythagorean triplet found: ", (a, b, c)
            result = max(result, a * b * c)
            
    print "max product is equal to ", result

